keps/0000-kep-template.md

```diff
@@ -37,15 +37,16 @@ The title should be lowercased and spaces/punctuation should be replaced with `-
 As the KEP is approved and an official KEP number is allocated, the file should be renamed.
 
 To get started with this template:
-* Make a copy in the appropriate directory.
+* Make a copy of this template.
   Name it `draft-YYYYMMDD-my-title.md`.
 * Create a PR in the [`kubernetes/community`](https://github.com/kubernetes/community) repo.
 * Check in early.
   Do this once the document holds together and general direction is understood by many in the sponsoring SIG.
   View anything marked as a draft as a working document.
   Aim for single topic PRs to keep discussions focused.
   If you disagree with what is already in a document, open a new PR with suggested changes.
-* As a KEP is approved, rename the file yet again with the final KEP number.
+* A KEP number can be assigned at any time by (a) taking the next number in the NEXT_KEP_NUMBER file and (b) incrementing that number.
+  These PRs should be approved quickly to minimize merge conflicts.
```

Comment:
The strict enumeration kind of seems like a #fail , why do we need strict ordering?

Reply:
If we do this we should just have a script that does this for us.

Reply:
The idea of assigning a number will help to (a) make these feel more official (b) show some ordering in terms of where these fit into the history of the project and (c) provide a way to unambiguously refer to a KEP.

RFCs and PEPs both have numbers and it helps bring the proposals together to feel more like a coherent whole, IMO.

I'd be open for figuring out how to have a script help out on the mechanics here.